src/EditDistPeriodicityDetector.cpp

```diff
@@ -55,60 +59,89 @@ namespace geopm
     {
         if (record.event == EVENT_REGION_ENTRY) {
             m_history_buffer.insert(record.signal);
+            ++m_record_count;
             calc_period();
         }
     }
 
+    void EditDistPeriodicityDetector::Dset(int ii, int jj, int mm, uint32_t val) {
+        m_DP[((ii % m_history_buffer_size)  * m_history_buffer_size
+            + (jj % m_history_buffer_size)) * m_history_buffer_size
+            + (mm % m_history_buffer_size)] = val;
+    }
+
+    uint32_t EditDistPeriodicityDetector::Dget(int ii, int jj, int mm) {
+        if (ii <= m_record_count - m_history_buffer_size) {
+            // This value is supposed to be INF but not so large that it gets wrapped around when
+            // a small value is added to it.
+            return std::numeric_limits<uint32_t>::max()/2;
+        }
+        if (jj >= m_history_buffer_size) {
+            // This value is supposed to be INF but not so large that it gets wrapped around when
+            // a small value is added to it.
+            return std::numeric_limits<uint32_t>::max()/2;
+        }
+        if (mm <= m_record_count - m_history_buffer_size) {
+            // This value is supposed to be INF but not so large that it gets wrapped around when
+            // a small value is added to it.
+            return std::numeric_limits<uint32_t>::max()/2;
+        }
+        return m_DP[((ii % m_history_buffer_size)  * m_history_buffer_size
```

Comment:
((ii % m_history_buffer_size) * m_history_buffer_size * m_history_buffer_size

Reply:
Wrote these indices more explicitly to easier understand.